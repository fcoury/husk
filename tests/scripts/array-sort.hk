// Test sort() method for arrays
let numbers = [3, 1, 4, 1, 5, 9, 2, 6];
let strings = ["zebra", "apple", "banana", "cherry"];
let bools = [true, false, true, false];
let empty = [];
let single = [42];

// Test number sorting
println(numbers.sort());     // Should print: [1, 1, 2, 3, 4, 5, 6, 9]

// Test string sorting
println(strings.sort());     // Should print: ["apple", "banana", "cherry", "zebra"]

// Test boolean sorting  
println(bools.sort());       // Should print: [false, false, true, true]

// Test empty array
println(empty.sort());       // Should print: []

// Test single element
println(single.sort());      // Should print: [42]

// Test negative numbers
let negatives = [-3, 5, -1, 0, 2];
println(negatives.sort());   // Should print: [-3, -1, 0, 2, 5]

// Test floats
let floats = [3.14, 2.71, 1.41, 2.0];
println(floats.sort());      // Should print: [1.41, 2.0, 2.71, 3.14]

// Test that original array is not modified (returns new array)
let original = [3, 1, 2];
let sorted = original.sort();
println(original);           // Should print: [3, 1, 2] (unchanged)
println(sorted);             // Should print: [1, 2, 3]

// Test chaining
println([5, 2, 8, 1].sort()); // Should print: [1, 2, 5, 8]

// Test Unicode strings
let unicode = ["café", "apple", "naïve", "zebra"];
println(unicode.sort());     // Should print: ["apple", "café", "naïve", "zebra"]