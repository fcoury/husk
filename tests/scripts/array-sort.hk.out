[1, 1, 2, 3, 4, 5, 6, 9]
[apple, banana, cherry, zebra]
[false, false, true, true]
[]
[42]
[-3, -1, 0, 2, 5]
[1.41, 2, 2.71, 3.14]
[3, 1, 2]
[1, 2, 3]
[1, 2, 5, 8]
[apple, café, naïve, zebra]
